rewriter/Util.cc

```diff
@@ -8,12 +8,20 @@ using namespace std;
 
 namespace sorbet::rewriter {
 
-ast::ExpressionPtr ASTUtil::dupType(const ast::ExpressionPtr &orig) {
+ast::ExpressionPtr ASTUtil::dupType(const ast::ExpressionPtr &orig, const ast::ExpressionPtr **failing) {
+#define RECORD_FAILING(tree)                  \
```

Comment:
😅 do we have to use a macro?

Reply:
It's not a requirement, no -- certainly it seems like Sorbet style is to shy away from macros.  But I think in this case it makes it clearer what's going on and easier to visually verify that the expression that you're recording as failing is the same expression that you attempted to call `dupType` on slightly earlier.

Reply:
> makes it clearer what's going on and easier to visually verify

would this not have been the case had it been implemented with a function?